### DISCUSSION TITLE
Change primary to worker and vice versa?

### DISCUSSION THREAD

**Discussion by tehniemer** [jhuckaby/Cronicle#760] [Q&A] [answered]
I want to switch the machine my primary server is on; can I demote the existing primary to a worker and promote one of the workers to the primary?

**Comment by jhuckaby** [comment, marked as answer]
Unfortunately, a primary server cannot be demoted while it is running.  Once a server becomes primary, it will remain primary until it shuts down.

So what you can do, theoretically, is change your server group configuration so that your worker is "eligible" to become primary, then shut down the current primary server.  That should cause a primary election to be held after 60 seconds, at which time the most eligible remaining server will be blessed (the one whose hostname alphabetizes first).  Then later you can restart your old primary server again.

Note that your worker needs to be "ready" to become a primary.  Specifically, it needs access to your storage system (S3 or other), and be configured just like your current primary is.

**Reply by tehniemer** [reply]
got it, thanks.
